api/src/main/java/org/eclipse/microprofile/jwt/algorithm/ContentEncryptionAlgorithm.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.algorithm;
+
+/**
+ * * JWT JSON Web Content Encryption Algorithms.
+ * 
+ * @see <a href="https://tools.ietf.org/html/rfc7518#section-5">https://tools.ietf.org/html/rfc7518#section-5</a>
+ */
+public enum ContentEncryptionAlgorithm {
+    A256GCM("A256GCM"),
+    A128CBC_HS256("A128CBC-HS256"),
+    A256CBC_HS512("A256CBC-HS512");
```

Comment:
Also here the list is missing encryption Methods if we go for an open spec. (and enum doesn't allow easy extension for new encryptions)

A128GCM, A192GCM, A192CBC-HS384

Reply:
See the comment below. This API is open enough, but has been carefully looked at. Lets focus on the quality of the API. Please take the latest smallrye library just to have a feel for it, IMHO it is quite good and will be good for MP JWT

Reply:
I find it an arbitrary decision which algorithms and encryption methods are supported on top of the one which can be used within MP JWT spec.

So in other words, if we only support those defined in the spec (like RS256 is the only mentioned) or all of them, that would make sense.

But now we have some of them and other not. (and not all of the JWA specified ones are now in the PR)

Reply:
And saying they can optionally is a bad thing. Implementation A will support all possible, while implementation B only the limited list.

This means an app which works on Implementation A fails on Implementation B which is an undesirable situation.

Reply:
@rdebusscher same is true for the line that other than 2048/1024 bits are supported optionally for the verification keys, right ?
That what I was implying. Saying a given algorithm is optional is a clear warning to the users that the interoperability is not guaranteed. But if a user has a requirement to create a token with something other than `RS256` using this API then they will have an option to use this API, assuming the given implementation is ready to do it. If this API does not have a typed support for a given algorithm then the users will need to open an issue and justify one more enum value added. I've already justified to you why `ES256` and explained why `HS256` is added (we have the users manually generating the cookies and this where the API should help) 

Anyway, I'll trim few values for now, `RS384`/`RS512`, etc, but let me say again, I don't agree that MP JWT server requirements are the ones that must drive what may be in the generation API, there is no such requirement and it is wrong IMHO to assert it. It is not a good strategy to keep a closed world assumption exactly for the same reason as assuming that soon all the world will start issuing MP JWT compliant tokens and they don't then it is not the problem of MP JWT.

Reply:
> So in other words, if we only support those defined in the spec (like RS256 is the only mentioned) or all of them, that would make sense.
I've several time already, no, IMHO it would not make sense to support all of them, and it was not the intention. I went through the list and chose few required/recommended ones (as well based on some evidence that those algorithms may be used in practice) and this enums must not include optional ones, and if some of them made it there then I'll trim them.